### DISCUSSION TITLE
Delete auto-created host users after they were deleted from Teleport, but not after ssh session

### DISCUSSION THREAD

**Discussion by pznamensky** [gravitational/teleport#22892] [Q&A] [answered]
Hey 👋
I'm sure you guys have a lot of experience with Teleport. Need advice.
Our developers have ssh access to the servers. From time to time they have to execute long-running tasks in tmux or screen.
Teleport has a great feature for auto-creating system users for each human user:
https://goteleport.com/docs/server-access/guides/host-user-creation/
Once the ssh session ends, Teleport will remove those users.

I have two concerns about this flow:
I'd like users to have the ability to execute long-running tasks in tmux/screen in the background without being killed. In other words, system users shouldn't be deleted after a session ends.
I'd like users to be deleted right after they were removed from Teleport (or Gsuite/GitHub/... if they are used as a SSO) no matter if they have an active ssh connection or not.

I know it sounds like good (or not) old LDAP 😄 
But do you have any ideas or thoughts about how to cook Teleport this way?

**Comment by webvictim** [comment, marked as answer]
I think that theoretically as long as you logged in via Teleport and spawned the `screen`/`tmux` process that way, the user should stay intact until the Teleport process officially exits i.e. `screen`/`tmux` terminates, which could potentially be days later. The main issue might be that if you restart Teleport on the machine, the Teleport process will be terminated or perhaps be reparented to PID 1, which means you'd lose track of the sessions.

You'd need some kind of out-of-band API callback process to delete the users from Teleport after removal from the SSO provider - this is basically what SCIM does, but it isn't supported by Teleport today. Also, removing a user from Teleport itself won't delete any OS logins associated with that user - you'd need some kind of separate cleanup/reaper process to do that for you.

Honestly, LDAP and `sssd` is probably a better fit for you. Teleport's host user creation is designed for short-lived ephemeral sessions which don't persist anything to disk. There are no UID/GID guarantees when it creates users, it just uses whatever the next one in the sequence is.

**Reply by pznamensky** [reply]
@webvictim thank you very much for your ideas and helpful answer!
I'll think about other options then.

**Reply by webvictim** [reply]
No problem. If longer-lived user management is something you'd like to see Teleport do, please feel free to create a feature request: https://github.com/gravitational/teleport/issues/new?assignees=&labels=feature-request&template=feature_request.md

We use feedback like this and comments/upvotes on issues to influence the future product roadmap.

**Reply by pznamensky** [reply]
Thanks!
